matching-core/src/main/java/com/graphhopper/matching/GPXExtension.java

```diff
@@ -17,36 +17,50 @@
  */
 package com.graphhopper.matching;
 
+import com.graphhopper.routing.VirtualEdgeIteratorState;
 import com.graphhopper.storage.index.QueryResult;
 import com.graphhopper.util.GPXEntry;
 
 /**
+ * Represents a map matching candidate.
  *
  * @author Peter Karich
+ * @author kodonnell
+ * @author Stefan Holder
  */
 public class GPXExtension {
-    final GPXEntry entry;
-    final QueryResult queryResult;
-    final int gpxListIndex;
+    public final GPXEntry entry;
+    public final QueryResult queryResult;
+    public final boolean isDirected;
+    public final VirtualEdgeIteratorState incomingVirtualEdge;
+    public final VirtualEdgeIteratorState outgoingVirtualEdge;
```

Comment:
I do not like having the virtual edges public, can we reduce visibility or use EdgeIteratorState instead?